### DISCUSSION TITLE
Ambiente de Desenvolvimento [S.O]

### DISCUSSION THREAD

**Discussion by romulo2735** [frontendbr/forum#454] [Discussão]
Uso windows para desenvolver aplicações academicas e até aplicações no meu trabalho, conheço muito pouco sobre SO's, mas muitos me dizem que o LINUX é otimo para desenvolvimento, a distribuição que me recomendaram foi a UBUNTU. queria saber a opnião de vcs que são mais experientes no ramo?

**Comment by robsonbittencourt** [comment]
Não manjo muito de linux e suas diferentes distribuições. Mas assim como
voçê utilizava Windows como SO para desenvolvimento, e troquei para Ubuntu.

Não lembro o motivo da escolha pois isso já faz algum tempo. Mas recomendo
que você faça está troca também. Instalar e manter coisas relativas ao
ambiente de desenvolvimento no Ubuntu é bem mais fácil que no Linux. E você
vai ficar mais amigo do terminal, o que pode aumentar a sua produtividade.

**Comment by guilhermejcgois** [comment]
Pessoalmente não curto muito ubuntu, em especial por causa do unity rsrs Mas se você não tem nenhuma experiência com linux, ubuntu é uma boa medida, recomendaria dar uma olhada no Mint também...

O legal de ubuntu e outros que usam o apt é que o repositório de aplicativos é bem generoso, e a maioria dos tutoriais de instalação que se vê por aqui (quase) sempre tem para ubuntu. Agora se você quer aprender a fazer as coisas na mão, configurar etc etc recomendaria o gentoo ou debian

**Comment by willycamargo** [comment]
Utilizei o Ubuntu por quase 5 anos (agora estou no OSX). Na situação em que comecei a utilizar o windows atual era o Windows 7, portanto era necessário fazer muita gambiarra para subir um ambiente de desenvolvimento, o que parecia ser simples no linux nos vídeos e artigos que via na época.

Inicialmente eu instalei em um dual boot com Ubuntu 8.04 e Windows 7. Principalmente porque utilizava muito pacote office e adobe (ministrava aulas), portanto só estava no Ubuntu quando realmente não precisa utilizar esses softwares.

Vale ressaltar aqui que na minha visão de nada adiantava instalar uma distribuição linux e tentar rodar pacote office ou adobe nativamente. Seria fugir de um problema para encontrar outros. Sempre acaba em gambiarra!

Com o tempo eu parei de ministrar aulas, portanto não estava precisando mais utilizar o pacote office e comecei a trabalhar mais com desenvolvimento web. Ainda tinha muita necessidade de utilizar photoshop e as vezes illustrator, portanto peguei uma imagem do windows XP muito leve com um amigo, subi ela no Virtual Box e rodei photoshop e illustrator. Deu certo!

Nessa época eu acho que já estava com o Ubuntu 10.10 (sempre mantive ele atualizado) e com todo o ambiente necessário rodando nele. Apesar de ter continuado com dual boot eu acredito que fiquei mais de 2 anos sem logar no windows. Utilizava o ubuntu como meu sistema principal.

O que quero passar aqui é que dificilmente você ai conseguir migrar de um sistema para o outro de forma muito rápida. Inclusive, esse choque pode acabar te traumatizando. IMHO, o ideal é não ser hipócrita, aceitar que depende de alguns softwares do Windows e montar um processo de migração. Analisar quais softwares você necessita e tentar substitui-los ou adptar para seu novo S.O.

Ainda vale ressaltar que não sei como está o windows atualmente. Principalmente com toda esse movimento da Microsoft em direção do mercado Open Source e essa parceria com Canonical. Talvez hoje seja tão simples subir um ambiente de desenvolvimento no windows que realmente não vale a pena ficar quebrando a cabeça com distribuições linux.

Anyway, cada caso é um caso. Desejo boa sorte.

**Comment by hugocarreira** [comment]
Sempre utilizei Linux para desenvolvimento, sempre os "Debian Based". Porém no início foi uma transição complicada, comecei virtualizando Ubuntu no meu Windows e depois de muito tempo arrisquei ficar só com o linux no PC, fiquei um tempinho utilizando só o ubuntu mas senti falta de algumas coisas que o windows me proporcionava ( joguinhos rs ) e acabei voltando pro SO do Bill Gates. 

Nessa transição de volta, não consegui me adaptar para desenvolver no windows, não conseguia fazer nada, sentia falta do terminal no meu dia a dia e os 'emuladores' existentes eram uma merda. Foi ai que resolvi fazer um dual-boot no meu computador, com isso meus problemas acabaram, era só reiniciar o pc e começar a desenvolver, foi glórias durante muito tempo.  

Atualmente eu tirei meu dual-boot por ter ganhado um notebook melhor e o utilizo totalmente focado para desenvolvimento, nele está instalado meu Ubuntu e digo que sou bem feliz e confortável programando em ambiente gnu/linux.

**Comment by annielmenezes** [comment]
@romulosousa27, eu utilizo Ubuntu no trabalho para desenvolvimento há uns 3 anos e no início foi complicado não conhecia na época o [Avocode](http://avocode.com) e dependia muito do Photoshop para fatiar os arquivos que recebia de design, tentei utilizar uma máquina virtual mas minha máquina física não era muito boa e tive que voltar para Windows por alguns meses. 

Atualmente eu utilizo o Ubuntu sem a VM, mas recentemente precisei trabalhar com um arquivo do Illustrator e voltei a ter problemas, no entanto, para resolver isso eu voltei ao Windows 10 de host com editores de imagens e afins e utilizo uma VM com o Ubuntu para desenvolvimento, como isso eu ganhei mais facilidade para testes crossbrowser, porque tenho o edge nativo e dois SO's

**Comment by romulo2735** [comment]
@willycamargo Eu uso o Windows 10,já tive contato com o UBUNTU, nunca tinha experimentado, mas gostei do SO. Meu foco e desenvolvimento web, php,html,css, frameworks, algumas aplicações JAVA e C++, aplicação mobile esse SO pode me proporcionar um ambiente propicio?

**Comment by romulo2735** [comment]
@annielmenezes eu não pretendo mexer com esse programas da ADOBE de imagens, não tem "dom" ou feeling para isso, mas nem tudo é como a gnt quer, então eu acabo as vezes utilizando.

**Comment by romulo2735** [comment]
Existe alguma documentação de como usar ele, ou tutoriais?

**Comment by willycamargo** [comment]
> romulosousa27 commented 2 hours ago
@willycamargo Eu uso o Windows 10,já tive contato com o UBUNTU, nunca tinha experimentado, mas gostei do SO. Meu foco e desenvolvimento web, php,html,css, frameworks, algumas aplicações JAVA e C++, aplicação mobile esse SO pode me proporcionar um ambiente propicio?

Com desenvolvimento Android eu não tive nenhum problema. Mas também não fiz nada muito grande. Com windows phone talvez tenha problemas e com iPhone com certeza terá problemas (provavelmente iPhone somente no Mac).

**Comment by femontanha** [comment]
Usei windows por um tempo, depois Linux, mas talvez por causa de época (IE, Photoshop e etc) acabava usando maquina virtual, o que enchia o saco. De uns 5 anos pra cá so usei Mac e não vejo o porque largar por enquanto. (é caro, sim)

**Comment by fabiomirandaa** [comment]
Eu uso a distribuição [Elementary OS](https://elementary.io/), que é baseada em Ubuntu, porém é muito mais leve e tem uma interface que me agrada muito mais(Lembra até a do MacOS). O bom é que ela tem uma interface leve e Clean e você pode fazer nela tudo que faria com o Ubuntu. 

Eu recomendo muito que você parta pro Linux, pois quando eu comecei no mundo do desenvolvimento eu tive um certo sofrimento por usar o Windows e ter alguns problemas na instalação de dependências e por depender muito de interfaces gráficas para fazer as coisas.  No linux você acaba sendo obrigado a aprender realmente as coisas e tem uma compatibilidade maior e uma facilidade grande pra instalação de dependências. Por exemplo, assim que formato uma máquina por qualquer motivo apenas rodo [ESSE](https://github.com/fabiomirandaa/my-helpers/blob/master/install-dependencies.sh) arquivo e ele baixa e instala tudo que eu preciso, como Java, Node, Android Studio e etc.    

Vale bem a pena migrar para o Linux, se você quer um ambiente estável para o desenvolvimento, e sem contar que no servidor sua aplicação provavelmente ficará num ambiente linux, então você pode já ter um Know-how de como usar um sistema linux e desenvolver ela num ambiente próximo ao do Servidor.

**Comment by rubenmarcus** [comment]
aqui na Printi eu como sou o unico que corta layouts, sou o unico que usa Windows.. nunca tive problema com o Vagrant.. só symlinks... Mas isso foi remediado com algumas configurações na VM.. 

Chegamos a tentar instalar o Mint, mas uso muito o Photoshop para mensurar layouts,e cortar assets também. O Avocode é perfeito pra isso, mas é pago. Não tem sentido em pagar ele, sendo que a empresa já paga licença da Adobe.

**Comment by rubenmarcus** [comment]
recentemente fui instalar e atualizar alguns modulos do npm na vm no vagrant, e tive problemas com o windows isso aqui ajudou : https://github.com/npm/npm/issues/7308#issuecomment-73489075

Mas nunca mais que isso...

**Comment by ranosi** [comment]
@fabiomirandaa eu to só esperando liberar meu note de algumas coisas pra trocar o ubuntu pelo ElementaryOs ttambém. Foi até bom ver esse teu resultado tão positivo com ele.
Ja vou dar uma copiada nesse teu script também com todo respeito rsrs

Sobre usar o Linux, também estou muito satisfeito, punica coisa que faço no windows é jogar mesmo o que não acontece a 3 meses rs. A curva que vejo pra quem chega no Linux assim do nada é enfrentar algumas dificuldades que exigem entrar no Google e buscar artigos mas sinceramente pra todo mundo aqui do grupo isso é uma coisa normal. Todos estão acostumados a estudar e buscar conteúdo.

**Comment by fabiomirandaa** [comment]
Pode copiar o script a vontade @ranosi . Se eu não me engano, dependendo do jogo, você pode jogar de boa com o Wine e Play On Linux, já tem um monte de jogos com a classificação Platinum no Wine(Comunidade gosta de jogar né hahaha). Quanto ao ElementaryOS eu acho muito bom, porém eu indico se puder, usar o Freya, que é baseado na Ubuntu 14.04 e por isso é o mais estável e com suporte até 2019. Eu to usando o Loki que é mais novo e tem apenas um bug que para alguns pode ser incomodo que é o botão de print que não funciona, mas como eu uso o app nativo para Screenshots não me incomoda.

**Comment by ranosi** [comment]
Vou confirmar a versão que baixei. Uso alguns programas com o Wine mas não cheguei a testar jogos. Mas como disse, jogar tem sido tão raro que nem é problema. Deve ter quase 3 meses que não uso um Windows.

**Comment by joffilyfe** [comment]
Eu particularmente não gosto muito do Ubuntu. Sou usuário do Linux desde 2003, não lembro bem a versão mas quando a interface gráfica foi migrada para o Unity o Ubuntu ficou muito mais lento. 

Atualmente eu uso o Debian em um Desktop e me atende 100%, também uso o OSX e não tenho o que reclamar entre os sistemas.

Pra quem é mais afoito pode tentar instalar o Arch Linux, ele funciona muitíssimo bem e é uma das melhores distribuições atuais.

**Comment by felipefialho** [comment]
Não sou muito ligado em SO pra ser sincero. Não tenho curiosidade para ficar testando e brincando com eles, nunca usei Linux por mais de 1h por exemplo.

Como muitos aqui sabem, usei o Windows 10 até meados do ano passado, quando fiquei 1 semana com um Macbook e acabei gostando muito, não resisti em comprar um, virando meu Laptop oficial. 

E sinceramente, o Mac OS é mais rápido e estável. Mas não tenho nada de muito grave para reclamar do Windows.

**Comment by ranosi** [comment]
Aproveitei um espaço no HD pra instalar o ElementaryOs no feriadão. Peguei alguns bugs nele que me deram uma certa desanimada, como o segundo monitor que não aceita maximizar nenhuma janela (quanto clico pra maximizar ele joga a janela pro canto). Apenas funcionou se eu deixar o monitor externo como principal. Deu bug no wallpaper tb que sumiu do nada e ficou uma tela parecido com televisão fora de sintonia rs.

Provavelmente vou instalar o Debian daqui umas semanas.

Na questão de trabalho acho que dá uma diferença boa em usar o Mac Os e Linux em comparação com Windows. E a vantagem do Mac é o sistema ser todo redondinho com o hardware né, ligou usou.

**Comment by adeonir** [comment]
Aqui na empresa é Windows e não tem escolha, mas em casa me aventurei no Hackintosh e depois de tudo configurado é só alegria.

Já no notebook, testei várias distros Linux: Elementary, Ubuntu, OpenSuse... A que mais gostei e uso até hoje é o Fedora. Package manager DNF simplesmente mais foda que APT-GET.
Recomendo!
